src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -633,11 +633,12 @@ private void logProvisionInfo(String message) {
             }
         }
 
+        net.setAssociatePublicIpAddress(false);
```

Comment:
```suggestion
        net.setAssociatePublicIpAddress(getAssociatePublicIp());
```
Will allow you to remove the if